### DISCUSSION TITLE
Does the SignalR Client automatically adjust the value of the KeepAliveInterval when the ServerTimeout value is changed?

### DISCUSSION THREAD

**Discussion by quytrieu37** [dotnet/aspnetcore#55126] [Q&A] [answered]
Hi SignalR team!
I've increased the ServerTimeout using the WithServerTimeout() method.
![image](https://github.com/dotnet/aspnetcore/assets/74439440/74a0f1ee-7a4f-412b-a094-681f755cf83e)
After a period of time typically around 2 to 5 minutes, the connection is closed with the following message:

> The server closed the connection with the following error: Connection closed with an error. OperationCanceledException: Client hasn't sent a message/ping within the configured ClientTimeoutInterval.

When I add `.WithKeepAliveInterval(TimeSpan.FromSeconds(15))`, The connection remains connected after 15 minutes. I understand that the default `KeepAliveInterval` value on the client side is set to 15 seconds. 
However, Does SignalR automatically adjust this value when the `ServerTimeout` value is changed?
I have not found any documentation addressing this aspect.

**Comment by davidfowl** [comment, marked as answer]
No, it does not.

**Comment by quytrieu37** [comment]
Thank you for your quick response.
